src/borg/archive.py

```diff
@@ -858,10 +858,13 @@ def same_item(item, st):
             st = os.stat(path, follow_symlinks=False)
             if continue_extraction and same_item(item, st):
                 return  # done! we already have fully extracted this file in a previous run.
-            elif stat.S_ISDIR(st.st_mode):
-                os.rmdir(path)
-            else:
+            # remove anything that is not a directory
+            if not stat.S_ISDIR(st.st_mode):
                 os.unlink(path)
+            # only remove a directory if it is conflicting
+            # preserve existing directories because they might be subvolumes
```

Comment:
is subvolumes a feature implemented by other fs than btrfs?

if not, maybe we better call it "btrfs subvolumes" here, so it is more clear.